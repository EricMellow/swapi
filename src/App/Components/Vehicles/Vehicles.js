import React from 'react';
import './Vehicles.css';
import PropTypes from 'prop-types';

const Vehicle = ({ cardData, toggleFavorite }) => {
  let vehicleCards = [];
  if (cardData) {
    vehicleCards = cardData.map((vehicle, index) => {
      return (
        <div key={vehicle.name + index} className="card">
          <h3>{vehicle.name}</h3>
          <button className="fav" onClick={() => toggleFavorite(vehicle)}></button>
          <p>Model: {vehicle.model}</p>
          <p>Class: {vehicle.vehicleClass}</p>
          <p>Number of Passengers: {vehicle.numOfPassengers}</p>
        </div>
      )
    })

  }

  return (
    <div className="container">
      {vehicleCards.length === 0 && <div className="wait">Access code YT-1300. Retreiving your SWAPIBOX experience. Please Wait.</div>}
      {vehicleCards.length > 1 && vehicleCards}
    </div>
  );
};

Vehicle.propTypes = {
  cardData: PropTypes.array,
  toggleFavorite: PropTypes.func
};

export default Vehicle;