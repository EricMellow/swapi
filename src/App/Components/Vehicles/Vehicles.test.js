import React from 'react';
import Vehicles from './Vehicles';
import { shallow } from 'enzyme';
import { mockCleanedVehicles } from '../../mockVehiclesData';

describe('Vehicles', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Vehicles />);

  });

  it('should match the snapshot with no data', () => {

    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot with good card data', () => {
    const mockData = mockCleanedVehicles;
    wrapper = shallow(<Vehicles cardData={mockData} />)

    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot with error data', () => {
    const mockError = [
      {
        model: "Something went wrong",
        name: "Oh no!",
        numOfPassengers: "No data",
        vehicleClass: ":("
      }
    ];
    wrapper = shallow(<Vehicles cardData={mockError}/>);

    expect(wrapper).toMatchSnapshot();

  });
});