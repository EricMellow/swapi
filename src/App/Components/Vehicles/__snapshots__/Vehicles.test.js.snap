// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vehicles should match the snapshot with error data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Vehicle
    cardData={
        Array [
            Object {
              "model": "Something went wrong",
              "name": "Oh no!",
              "numOfPassengers": "No data",
              "vehicleClass": ":(",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        false,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      false,
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          false,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        false,
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Vehicles should match the snapshot with good card data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Vehicle
    cardData={
        Array [
            Object {
              "model": "Digger Crawler",
              "name": "Sand Crawler",
              "passengers": "30",
              "vehicleClass": "wheeled",
            },
            Object {
              "model": "Trans Am",
              "name": "Kitt",
              "passengers": "2",
              "vehicleClass": "Knight Rider",
            },
            Object {
              "model": "Charger",
              "name": "General Lee",
              "passengers": "2",
              "vehicleClass": "wheeled",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        Array [
          <div
            className="card"
>
            <h3>
                        Sand Crawler
            </h3>
            <button
                        className="fav"
                        onClick={[Function]}
            />
            <p>
                        Model: 
                        Digger Crawler
            </p>
            <p>
                        Class: 
                        wheeled
            </p>
            <p>
                        Number of Passengers: 
            </p>
</div>,
          <div
            className="card"
>
            <h3>
                        Kitt
            </h3>
            <button
                        className="fav"
                        onClick={[Function]}
            />
            <p>
                        Model: 
                        Trans Am
            </p>
            <p>
                        Class: 
                        Knight Rider
            </p>
            <p>
                        Number of Passengers: 
            </p>
</div>,
          <div
            className="card"
>
            <h3>
                        General Lee
            </h3>
            <button
                        className="fav"
                        onClick={[Function]}
            />
            <p>
                        Model: 
                        Charger
            </p>
            <p>
                        Class: 
                        wheeled
            </p>
            <p>
                        Number of Passengers: 
            </p>
</div>,
        ],
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      false,
      Object {
        "instance": null,
        "key": "Sand Crawler0",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              Sand Crawler
</h3>,
            <button
              className="fav"
              onClick={[Function]}
/>,
            <p>
              Model: 
              Digger Crawler
</p>,
            <p>
              Class: 
              wheeled
</p>,
            <p>
              Number of Passengers: 
</p>,
          ],
          "className": "card",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sand Crawler",
            },
            "ref": null,
            "rendered": "Sand Crawler",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fav",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Model: ",
                "Digger Crawler",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Model: ",
              "Digger Crawler",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Class: ",
                "wheeled",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Class: ",
              "wheeled",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Number of Passengers: ",
                undefined,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Number of Passengers: ",
              undefined,
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "Kitt1",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              Kitt
</h3>,
            <button
              className="fav"
              onClick={[Function]}
/>,
            <p>
              Model: 
              Trans Am
</p>,
            <p>
              Class: 
              Knight Rider
</p>,
            <p>
              Number of Passengers: 
</p>,
          ],
          "className": "card",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Kitt",
            },
            "ref": null,
            "rendered": "Kitt",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fav",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Model: ",
                "Trans Am",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Model: ",
              "Trans Am",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Class: ",
                "Knight Rider",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Class: ",
              "Knight Rider",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Number of Passengers: ",
                undefined,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Number of Passengers: ",
              undefined,
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "General Lee2",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              General Lee
</h3>,
            <button
              className="fav"
              onClick={[Function]}
/>,
            <p>
              Model: 
              Charger
</p>,
            <p>
              Class: 
              wheeled
</p>,
            <p>
              Number of Passengers: 
</p>,
          ],
          "className": "card",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "General Lee",
            },
            "ref": null,
            "rendered": "General Lee",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fav",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Model: ",
                "Charger",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Model: ",
              "Charger",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Class: ",
                "wheeled",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Class: ",
              "wheeled",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Number of Passengers: ",
                undefined,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Number of Passengers: ",
              undefined,
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          Array [
            <div
              className="card"
>
              <h3>
                            Sand Crawler
              </h3>
              <button
                            className="fav"
                            onClick={[Function]}
              />
              <p>
                            Model: 
                            Digger Crawler
              </p>
              <p>
                            Class: 
                            wheeled
              </p>
              <p>
                            Number of Passengers: 
              </p>
</div>,
            <div
              className="card"
>
              <h3>
                            Kitt
              </h3>
              <button
                            className="fav"
                            onClick={[Function]}
              />
              <p>
                            Model: 
                            Trans Am
              </p>
              <p>
                            Class: 
                            Knight Rider
              </p>
              <p>
                            Number of Passengers: 
              </p>
</div>,
            <div
              className="card"
>
              <h3>
                            General Lee
              </h3>
              <button
                            className="fav"
                            onClick={[Function]}
              />
              <p>
                            Model: 
                            Charger
              </p>
              <p>
                            Class: 
                            wheeled
              </p>
              <p>
                            Number of Passengers: 
              </p>
</div>,
          ],
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": "Sand Crawler0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                Sand Crawler
</h3>,
              <button
                className="fav"
                onClick={[Function]}
/>,
              <p>
                Model: 
                Digger Crawler
</p>,
              <p>
                Class: 
                wheeled
</p>,
              <p>
                Number of Passengers: 
</p>,
            ],
            "className": "card",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sand Crawler",
              },
              "ref": null,
              "rendered": "Sand Crawler",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fav",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Model: ",
                  "Digger Crawler",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Model: ",
                "Digger Crawler",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Class: ",
                  "wheeled",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Class: ",
                "wheeled",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Number of Passengers: ",
                  undefined,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Number of Passengers: ",
                undefined,
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "Kitt1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                Kitt
</h3>,
              <button
                className="fav"
                onClick={[Function]}
/>,
              <p>
                Model: 
                Trans Am
</p>,
              <p>
                Class: 
                Knight Rider
</p>,
              <p>
                Number of Passengers: 
</p>,
            ],
            "className": "card",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Kitt",
              },
              "ref": null,
              "rendered": "Kitt",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fav",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Model: ",
                  "Trans Am",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Model: ",
                "Trans Am",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Class: ",
                  "Knight Rider",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Class: ",
                "Knight Rider",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Number of Passengers: ",
                  undefined,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Number of Passengers: ",
                undefined,
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "General Lee2",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                General Lee
</h3>,
              <button
                className="fav"
                onClick={[Function]}
/>,
              <p>
                Model: 
                Charger
</p>,
              <p>
                Class: 
                wheeled
</p>,
              <p>
                Number of Passengers: 
</p>,
            ],
            "className": "card",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "General Lee",
              },
              "ref": null,
              "rendered": "General Lee",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fav",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Model: ",
                  "Charger",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Model: ",
                "Charger",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Class: ",
                  "wheeled",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Class: ",
                "wheeled",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Number of Passengers: ",
                  undefined,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Number of Passengers: ",
                undefined,
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Vehicles should match the snapshot with no data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Vehicle />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="wait"
>
          Access code YT-1300. Retreiving your SWAPIBOX experience. Please Wait.
</div>,
        false,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Access code YT-1300. Retreiving your SWAPIBOX experience. Please Wait.",
          "className": "wait",
        },
        "ref": null,
        "rendered": "Access code YT-1300. Retreiving your SWAPIBOX experience. Please Wait.",
        "type": "div",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="wait"
>
            Access code YT-1300. Retreiving your SWAPIBOX experience. Please Wait.
</div>,
          false,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Access code YT-1300. Retreiving your SWAPIBOX experience. Please Wait.",
            "className": "wait",
          },
          "ref": null,
          "rendered": "Access code YT-1300. Retreiving your SWAPIBOX experience. Please Wait.",
          "type": "div",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
